src/librustc_mir/borrow_check/nll/type_check/liveness/point_index_map.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use borrow_check::nll::region_infer::values::PointIndex;
+use borrow_check::nll::region_infer::values::RegionValueElements;
+use rustc::mir::{BasicBlock, Location, Mir};
+use rustc_data_structures::indexed_vec::{Idx, IndexVec};
+use std::rc::Rc;
+
+/// A little data structure that makes it more efficient to find the
+/// predecessors of each point.
+crate struct PointIndexMap<'me, 'tcx> {
```

Comment:
oh, hmm, I had thought about removing this code and merging this into `RegionValueElements`. Maybe I should do that before landing.